File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/Cake.java
package nl.craftsmen.workshops.reactivemeetup.domain;

import java.util.ArrayList;
import java.util.List;

public class Cake {

    private final String name;
    private final List<Ingredient> ingredients = new ArrayList<>();

    public Cake(String name) {
        this.name = name;
    }

    public void addIngredient(Ingredient ingredient) {
        ingredients.add(ingredient);
    }

    public String toString() {
        StringBuilder sb = new StringBuilder(name).append(": ");
        for (Ingredient ingredient : ingredients) {
            sb.append(ingredient.getName().toLowerCase()).append(" ");
        }
        return sb.toString();
    }
}

File: javascript/src/domain/cake.js
var method = Cake.prototype;

var ingredient = require('./ingredient');

function Cake(name) {
    this._name = name;
    this._ingredients = [];
}

method.getName = function() {
    return this._name;
};

method.addIngredient = function(ingredient) {
        this._ingredients.push(ingredient);
}

method.getIngredients = function() {
    for (var ingredient in this._ingredients) {
         console.log(ingredient.name);
    }
}

module.exports = Cake;